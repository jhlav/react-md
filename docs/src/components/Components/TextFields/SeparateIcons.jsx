import React from 'react';
import {
  FontIcon,
  SVGIcon,
  TextField,
} from 'react-md';

import invisible from 'icons/visibility_off.svg';
import visible from 'icons/visibility.svg';
import locked from 'icons/lock.svg';
import unlocked from 'icons/lock_open.svg';

const SeparateIcons = () => (
  <div className="md-grid text-fields">
    <TextField
      className="md-cell md-cell--bottom"
      id="password-field-two-icons-with-font-icon"
      label="Password with FontIcon"
      passwordIcon={{
        invisible: <FontIcon>visibility_off</FontIcon>,
        visible: <FontIcon>visibility</FontIcon>,
      }}
      type="password"
    />
    <TextField
      className="md-cell md-cell--bottom"
      id="password-field-two-icons-with-svg-icon"
      label="Password with SVG"
      passwordIcon={{
        invisible: <SVGIcon use={invisible.url} />,
        visible: <SVGIcon use={visible.url} />,
      }}
      type="password"
    />
    <TextField
      className="md-cell md-cell--bottom"
      id="password-field-two-icons-with-alt-svg-icon"
      label="Password with Alternate SVG"
      passwordIcon={{
        invisible: <SVGIcon use={locked.url} />,
        visible: <SVGIcon use={unlocked.url} />,
      }}
      type="password"
    />
  </div>
);
export default SeparateIcons;
