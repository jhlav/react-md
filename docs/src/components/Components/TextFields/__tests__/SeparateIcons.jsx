/* eslint-env jest */
import React from 'react';
import { render } from 'enzyme';

import SeparateIcons from '../SeparateIcons';

describe('SeparateIcons', () => {
  it('should render correctly', () => {
    const tree = render(<SeparateIcons />);
    expect(tree).toMatchSnapshot();
  });
});
