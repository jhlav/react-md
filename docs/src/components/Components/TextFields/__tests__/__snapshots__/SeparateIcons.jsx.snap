// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeparateIcons should render correctly 1`] = `
<div
  class="md-grid text-fields"
>
  <div
    class="md-text-field-container md-full-width md-text-field-container--input md-cell md-cell--bottom"
  >
    <label
      class="md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary"
      for="password-field-two-icons-with-font-icon"
    >
      Password with FontIcon
    </label>
    <input
      class="md-text-field md-text-field--inline-indicator md-text-field--floating-margin md-full-width md-text"
      id="password-field-two-icons-with-font-icon"
      type="password"
    />
    <hr
      class="md-divider md-divider--text-field md-divider--expand-from-left"
    />
    <button
      class="md-text-field-inline-indicator md-password-btn md-pointer--hover md-text-field-inline-indicator--floating md-text--disabled"
      type="button"
    >
      <i
        class="md-icon material-icons"
      >
        visibility_off
      </i>
    </button>
  </div>
  <div
    class="md-text-field-container md-full-width md-text-field-container--input md-cell md-cell--bottom"
  >
    <label
      class="md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary"
      for="password-field-two-icons-with-svg-icon"
    >
      Password with SVG
    </label>
    <input
      class="md-text-field md-text-field--inline-indicator md-text-field--floating-margin md-full-width md-text"
      id="password-field-two-icons-with-svg-icon"
      type="password"
    />
    <hr
      class="md-divider md-divider--text-field md-divider--expand-from-left"
    />
    <button
      class="md-text-field-inline-indicator md-password-btn md-pointer--hover md-text-field-inline-indicator--floating md-text--disabled"
      type="button"
    >
      <svg
        class="md-icon"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      />
    </button>
  </div>
  <div
    class="md-text-field-container md-full-width md-text-field-container--input md-cell md-cell--bottom"
  >
    <label
      class="md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary"
      for="password-field-two-icons-with-alt-svg-icon"
    >
      Password with Alternate SVG
    </label>
    <input
      class="md-text-field md-text-field--inline-indicator md-text-field--floating-margin md-full-width md-text"
      id="password-field-two-icons-with-alt-svg-icon"
      type="password"
    />
    <hr
      class="md-divider md-divider--text-field md-divider--expand-from-left"
    />
    <button
      class="md-text-field-inline-indicator md-password-btn md-pointer--hover md-text-field-inline-indicator--floating md-text--disabled"
      type="button"
    >
      <svg
        class="md-icon"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      />
    </button>
  </div>
</div>
`;
